### ISSUE TITLE
Evaluate caffe model uploaded in caffe model zoo

### ISSUE THREAD

**Issue by liaocs2008** [wenwei202/caffe#5]
### Issue summary
Evaluating your uploaded models in general caffe (https://github.com/BVLC/caffe/tree/master/src/caffe) results in following:

> I0503 16:43:44.044796 19898 caffe.cpp:155] Finetuning from /home/leo/Downloads/**caffenet_SSL_0.4259.caffemodel**
I0503 16:43:48.726794 19898 caffe.cpp:251] Starting Optimization
I0503 16:43:48.726846 19898 solver.cpp:279] Solving AlexNet
I0503 16:43:48.726857 19898 solver.cpp:280] Learning Rate Policy: step
I0503 16:43:48.751430 19898 solver.cpp:337] Iteration 0, Testing net (#0)
I0503 16:43:49.218724 19898 blocking_queue.cpp:50] Data layer prefetch queue empty
I0503 16:44:20.341884 19898 solver.cpp:404]     Test net output #0: loss = 2.91344 (* 1 = 2.91344 loss)
I0503 16:44:20.342052 19898 solver.cpp:404]     Test net output #1: top1_accuracy = 0.40986
I0503 16:44:20.342062 19898 solver.cpp:404]     Test net output #2: top5_accuracy = 0.658339

> I0503 16:51:35.119240 26706 caffe.cpp:155] Finetuning from /home/leo/Downloads/**caffenet_SSL_0.4469.caffemodel**
I0503 16:51:43.908475 26706 caffe.cpp:251] Starting Optimization
I0503 16:51:43.908524 26706 solver.cpp:279] Solving AlexNet
I0503 16:51:43.908535 26706 solver.cpp:280] Learning Rate Policy: step
I0503 16:51:43.916864 26706 solver.cpp:337] Iteration 0, Testing net (#0)
I0503 16:51:44.117143 26706 blocking_queue.cpp:50] Data layer prefetch queue empty
I0503 16:52:13.693686 26706 solver.cpp:404]     Test net output #0: loss = 2.72754 (* 1 = 2.72754 loss)
I0503 16:52:13.693859 26706 solver.cpp:404]     Test net output #1: top1_accuracy = 0.43084
I0503 16:52:13.693868 26706 solver.cpp:404]     Test net output #2: top5_accuracy = 0.67862

> I0503 16:50:10.291841 25409 caffe.cpp:155] Finetuning from /home/leo/Downloads/**caffenet_L1_0.4251.caffemodel**
I0503 16:50:12.619544 25409 caffe.cpp:251] Starting Optimization
I0503 16:50:12.619585 25409 solver.cpp:279] Solving AlexNet
I0503 16:50:12.619592 25409 solver.cpp:280] Learning Rate Policy: step
I0503 16:50:12.626791 25409 solver.cpp:337] Iteration 0, Testing net (#0)
I0503 16:50:12.895716 25409 blocking_queue.cpp:50] Data layer prefetch queue empty
I0503 16:50:50.041647 25409 solver.cpp:404]     Test net output #0: loss = 3.26702 (* 1 = 3.26702 loss)
I0503 16:50:50.041822 25409 solver.cpp:404]     Test net output #1: top1_accuracy = 0.36774
I0503 16:50:50.041837 25409 solver.cpp:404]     Test net output #2: top5_accuracy = 0.61448

> I0503 22:01:39.975592  8616 caffe.cpp:155] Finetuning from /home/leo/Downloads/**caffenet_L1_0.4467.caffemodel**
I0503 22:01:46.399006  8616 caffe.cpp:251] Starting Optimization
I0503 22:01:46.399055  8616 solver.cpp:279] Solving AlexNet
I0503 22:01:46.399062  8616 solver.cpp:280] Learning Rate Policy: step
I0503 22:01:46.405417  8616 solver.cpp:337] Iteration 0, Testing net (#0)
I0503 22:01:47.340291  8616 blocking_queue.cpp:50] Data layer prefetch queue empty
I0503 22:02:16.862293  8616 solver.cpp:404]     Test net output #0: loss = 2.76222 (* 1 = 2.76222 loss)
I0503 22:02:16.862720  8616 solver.cpp:404]     Test net output #1: top1_accuracy = 0.42278
I0503 22:02:16.862746  8616 solver.cpp:404]     Test net output #2: top5_accuracy = 0.673259

### Steps to reproduce
1. download models uploaded in caffe model zoo (https://github.com/BVLC/caffe/wiki/Model-Zoo#learning-structured-sparsity-in-deep-neural-networks)
2. configure train_val.prototxt as following:

layer
 {
  name: "top5_accuracy"
  type: "Accuracy"
  bottom: "fc8"
  bottom: "label"
  top: "top5_accuracy"
  accuracy_param {
    top_k: 5
  }
  include {
    phase: TEST
  }
}
layer {
  name: "top1_accuracy"
  type: "Accuracy"
  bottom: "fc8"
  bottom: "label"
  top: "top1_accuracy"
  include {
    phase: TEST
  }
}

To make sure imagenet data is well prepared, I evaluate the alexnet model (https://github.com/BVLC/caffe/tree/master/models/bvlc_alexnet) and get following:

> I0503 16:45:40.494272 21305 blocking_queue.cpp:50] Data layer prefetch queue empty
I0503 16:46:14.405321 21305 solver.cpp:404]     Test net output #0: loss = 1.86402 (* 1 = 1.86402 loss)
I0503 16:46:14.405730 21305 solver.cpp:404]     Test net output #1: top1_accuracy = 0.56822
I0503 16:46:14.405750 21305 solver.cpp:404]     Test net output #2: top5_accuracy = 0.799561

### Your system configuration
Operating system: ubuntu 16.04
Compiler: 
CUDA version (if applicable): 8.0
CUDNN version (if applicable): 
BLAS:
Python or MATLAB version (for pycaffe and matcaffe respectively):

**Comment by wenwei202** [comment]
@liaocs2008 I notice you were using `AlexNet`. The uploaded model is `CaffeNet`. Please use the prototxt in `models/bvlc_reference_caffenet` to evaluate. Let me know if this is the issue. Tutorial is updated to avoid confusion. Thanks

**Comment by wenwei202** [comment]
@liaocs2008 Is it the issue? I will close this issue if it is.
